nav2_map_server/include/nav2_map_server/map_3d/map_saver_3d.hpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2020 Shivam Pandey
+// Copyright (c) 2020 Samsung Research Russia
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_MAP_SERVER__MAP_3D__MAP_SAVER_3D_HPP_
+#define NAV2_MAP_SERVER__MAP_3D__MAP_SAVER_3D_HPP_
+
+#include <string>
+#include <memory>
+
+#include "nav2_msgs/srv/save_map3_d.hpp"
+
+#include "nav2_map_server/map_saver_core.hpp"
+#include "nav2_map_server/map_3d/map_io_3d.hpp"
+
+namespace nav2_map_server
+{
+
+/**
+ * @class nav2_map_server::MapSaver<sensor_msgs::msg::PointCloud2>
+ * @brief A class that provides map saving methods and services
+ * for PointCloud maps, via. SaveMap3D service
+ * SaveMap service default name : save_map
+ */
+template<>
```

Comment:
Ditto templating -- I'll stop calling it out here since I think we get the pattern of these comments. The  savers/servers don't need this additioanl templating unless you're templating the MsgT and ServiceT to support both 2D occ grids and 3D pointclouds in a single object (which I'm not pushing for, but would be the only way that seems to make sense to me)